reana_server/reana_admin/check_workflows.py

```diff
@@ -91,56 +96,74 @@ def _collect_messages_from_scheduler_queue(filtered_workflows: Query) -> Dict:
     return scheduler_messages_collector.messages
 
 
-def _message_is_in_scheduler_queue(workflow, pods, scheduler_messages):
+def _message_is_in_scheduler_queue(
+    workflow: Workflow, pods: List[V1Pod], scheduler_messages: Dict
+):
     if str(workflow.id_) not in scheduler_messages:
         raise CheckFailed("Message is not found in workflow-submission queue.")
 
 
-def _pods_dont_exist(workflow, pods, scheduler_messages):
+def _pods_dont_exist(workflow: Workflow, pods: List[V1Pod], scheduler_messages: Dict):
     if len(pods) > 0:
         raise CheckFailed("Some pods still exist.")
 
 
-def _pods_exist(workflow, pods, scheduler_messages):
+def _pods_exist(workflow: Workflow, pods: List[V1Pod], scheduler_messages: Dict):
     if len(pods) == 0:
         raise CheckFailed("No pods found.")
 
 
-def _only_one_pod_exists(workflow, pods, scheduler_messages):
+def _only_one_pod_exists(
+    workflow: Workflow, pods: List[V1Pod], scheduler_messages: Dict
+):
     if len(pods) != 1:
         raise CheckFailed(f"Only one pod should exist. Found {len(pods)}.")
 
 
-def _all_pods_have_phase(workflow, pods, scheduler_messages, phase):
+def _all_pods_have_phase(
+    workflow: Workflow, pods: List[V1Pod], scheduler_messages: Dict, phase
+):
     if any([pod.status.phase != phase for pod in pods]):
         raise CheckFailed(f"Some pods are not in {phase}.")
 
 
-def _no_batch_pods_are_in_notready_state(workflow, pods, scheduler_messages):
+def _no_batch_pods_are_in_notready_state(
+    workflow: Workflow, pods: List[V1Pod], scheduler_messages: Dict
+):
     for pod in pods:
         for container in pod.status.container_statuses:
             if container.state.terminated:
                 raise CheckFailed(f"{pod.metadata.name} pod is in NotReady state.")
 
 
+def _workspace_exists(workflow: Workflow, pods: List[V1Pod], scheduler_messages: Dict):
+    if not Path(workflow.workspace_path).exists():
+        raise CheckFailed(f"The workspace '{workflow.workspace_path}' does not exist.")
+
+
 validation_map = {
```

Comment:
I did not think about stopped workflows, I will add the check for them! Regarding deleted workflows, it's been only ~1y since it has become mandatory to delete the workspace at the same time as the workflow, see https://github.com/reanahub/reana-client/issues/612. So it can happen that some workflows were deleted in the past but their workspace wasn't, so there is no way to know if  the workspace should be present on disk on not.

Reply:
I have added the check also for created workflows